test/plug/amqp/telemetry_test.exs

```diff
@@ -0,0 +1,95 @@
+defmodule Plug.AMQP.TelemetryTest do
+  use ExUnit.Case, async: true
+
+  def attach_telemetry do
+    unique_name = :"PID#{System.unique_integer()}"
```

Comment:
Why do you need to care about the name/registration. Why not just use a closure:

```
handler = self()
```
 
and then:

```
send(handler, {:event, event, measurements, metadata})
```

Reply:
It does not work, the telemetry events leak across tests.

Reply:
Actually my version is flaky too. Telemetry events from other test files are being captured by this handler.